tracing/tracer.go

```diff
@@ -0,0 +1,103 @@
+package tracing
+
+import (
+	"fmt"
+
+	"github.com/opentracing/opentracing-go"
+	zipkinot "github.com/openzipkin-contrib/zipkin-go-opentracing"
+	zipkin "github.com/openzipkin/zipkin-go"
+	"github.com/openzipkin/zipkin-go/reporter"
+	zipkinhttp "github.com/openzipkin/zipkin-go/reporter/http"
+	"github.com/reddit/baseplate.go/log"
+)
+
+var zipkinReporter reporter.Reporter
+
+// Tracer is exported to support start tracible server etc
+var Tracer = opentracing.GlobalTracer()
+
+// InitTracer configure the global tracer
+func InitTracer(
+	serviceName, addr string,
+	sampleRate float64,
+	logger log.Wrapper,
+) {
```

Comment:
The old `log.Fatalf` will also call panic, so now without that we do need to return an error from this function when something bad happens, so the caller would know to call panic.

Reply:
Yea, I was hoping we will extend existing logger.Wrapper to panic rather than return error every time we call logger.

Reply:
not all log messages should cause panic (e.g. the info log you are adding at the end, or the logging you added in `EndSpan` call), so that really should be handled by the caller instead of the library code.

Reply:
What I'm trying to say is that we need to add different log levels to the logger, so we can log info, debug, error, fatal.

This specific function is called during start up, and I think it makes sense to fail with non-zero status code.

Reply:
Yes but that means:
1. You need to proper document it (in the function comment) that the passed in log wrapper is supposed to call panic
2. That's kinda anti-pattern and error-prone (user who don't read the documentation carefully will likely be unaware of that)

Alternatively, you can just call panic after calling `logger`.

Also as a library code we really don't need info and debug level logs in most cases. Those are more suitable for service/application code.

In this case (`InitTracer` function), I don't really see why we can't just return the err to the caller without any logging, and the caller just do:

```
if err := tracing.InitLogger(...); err != nil {
  log.Fatalf(err)
}
```